torch_geometric/explain/glg_explainer.py

```diff
@@ -0,0 +1,621 @@
+from typing import Dict, Optional, Tuple
+from collections import defaultdict
+import math
+
+import torch
+from sympy import to_dnf, lambdify
+from torch import Tensor
+from torch_geometric.explain.metric import (
+    formulas_accuracy
+)
+from torch_geometric.explain import LogicExplanation, LogicExplanations
+from torch_geometric.loader import DataLoader
+from torch_scatter import scatter
+import torch.nn.functional as F
+
+
+class GLGExplainer(torch.nn.Module):
```

Comment:
Can we have this explainer inherit from `ExplainerAlgorithm` as other algorithms in [torch_geometric/explain/algorithm](https://github.com/pyg-team/pytorch_geometric/tree/master/torch_geometric/explain/algorithm) does? And have it conform to the same interface as other `ExplainerAlgorithm`s .

Reply:
Thanks for your comment. At first I actually tried to conform better to the interface, but then I realized that the differences between the proposed approach and the other Explainers available are substantial. Indeed, `ExplainerAlgorithm` seems heavily tailored for supporting Local Explainers (aka Instance-level Explainers), while the proposed one is a Global Explainer, and as such is radically different from the already implemented explainers.

For example, the forward function in `ExplainerAlgorithm` is meant to extract the explanations for every input instance, while for `GLGExplainer` the explanations are extracted during training, then you can simply inspect what the model has learned.

So, after chatting with @rusty1s, I opted for differentiating the two approaches, in order to avoid confusion about different frameworks sharing the same functions and the same input parameters (Local Explainers take as input graphs, while Global takes as input graphs which are actually local explanations)